molpipeline/mol2any/mol2rdkit_phys_chem.py

```diff
@@ -149,8 +176,10 @@ def set_params(self, **parameters: dict[str, Any]) -> Self:
             Self
         """
         parameters_shallow_copy = dict(parameters)
-        descriptor_list = parameters_shallow_copy.pop("descriptor_list", None)
-        if descriptor_list is not None:
-            self._descriptor_list = descriptor_list  # type: ignore
+        params_list = ["descriptor_list", "return_with_errors", "log_exceptions"]
+        for param_name in params_list:
+            if param_name in parameters:
+                setattr(self, f"_{param_name}", parameters[param_name])
```

Comment:
here we dont check the descriptor name. Could be solved with a property.

Reply:
You're right. I added a setter that validates the user-defined input data.